pkg/analysis/passes/codediff/codediff.go

```diff
@@ -99,3 +162,97 @@ func run(pass *analysis.Pass) (interface{}, error) {
 
 	return nil, nil
 }
+
+func generatePrompt(newVersion, newCommit, currentVersion, currentCommit string) (string, error) {
+	if newVersion == "" {
+		return "", errors.New("new version is empty")
+	}
+	if newCommit == "" {
+		return "", errors.New("new commit is empty")
+	}
+	if currentVersion == "" {
+		return "", errors.New("current version is empty")
+	}
+	if currentCommit == "" {
+		return "", errors.New("current commit is empty")
+	}
+
+	// Build questions section from llmreview questions
+	var questionsSection strings.Builder
+	for _, q := range llmreview.Questions {
+		questionsSection.WriteString("* ")
+		questionsSection.WriteString(q.Question)
+		questionsSection.WriteString("\n")
+	}
+
+	tmpl, err := template.New("prompt").Parse(promptTemplate)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse prompt template: %w", err)
+	}
+
+	data := map[string]any{
+		"NewVersion":     newVersion,
+		"NewCommit":      newCommit,
+		"CurrentVersion": currentVersion,
+		"CurrentCommit":  currentCommit,
+		"Questions":      strings.TrimSuffix(questionsSection.String(), "\n"),
+		"QuestionCount":  len(llmreview.Questions),
+	}
+
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, data); err != nil {
+		return "", fmt.Errorf("failed to execute prompt template: %w", err)
+	}
+
+	return buf.String(), nil
+}
+
+func runLLMAnalysis(
+	newVersion, newCommit, currentVersion, currentCommit, repositoryPath string,
+) ([]LLMAnalysisResponse, error) {
+	// Generate the prompt with dynamic version/commit information
+	prompt, err := generatePrompt(newVersion, newCommit, currentVersion, currentCommit)
+	if err != nil {
+		logme.Debugln("Failed to generate prompt:", err)
+		return nil, err
+	}
+
+	// clean up files from repositoryPath
+	cleanFiles := []string{"replies.json", ".nvmrc", "GEMINI.MD"}
+	for _, file := range cleanFiles {
+		filePath := filepath.Join(repositoryPath, file)
+		if _, err := os.Stat(filePath); err == nil {
```

Comment:
The err returned by `os.Stat` should be handled as well

Reply:
This error will trigger very often so I prefer we don't log it to reduce noise